src/views/BlogPostsList.vue

```diff
@@ -0,0 +1,119 @@
+<template>
+  <div class="page-container">
+    <div class="header-container">
+      <UserSwitcher :current-user="currentUser" :user-id="currentUser.userId" />
+      <button v-on:click="addNewPost" class="add-new-post-button">New Post</button>
+    </div>
+
+    <div class="content-container">
+      <h1 class="title-text">Blog posts</h1>
+
+      <BlogPostListMenu
+        :active-item="postsType"
+        :menu-items="['All', 'My Posts']"
+        :on-press-menu-item="changePostsType"
+      />
+
+      <div class="posts-list-container">
+        <BlogPostCard
+          v-for="post in postsType === 'All' ? allPosts : userPosts"
+          :post="post"
+          :current-user-id="currentUser.userId"
+          :key="post.postId"
+        />
+      </div>
```

Comment:
How are large lists in Vue typically optimised? For example, specifying the number of mounted `BlogPostCard`s at a given time

I know the Response cards in the dashboard is paginated, but has the app used an infinite scroll functionality somewhere else?

Reply:
- lazy loading
- pagination

Mobile -> responses (infinite scroll)